pkg/api/server.go

```diff
@@ -253,6 +267,9 @@ func (s *Server) startHTTP() error {
 
 	tracing.GoPanicWrap(s.ctx, &s.wg, "http", func(ctx context.Context) {
 		s.Log.Info("serving http", zap.String("address", s.httpListener.Addr().String()))
+		if s.httpListener == nil {
```

Comment:
The race test was catching on this. Surprised that wasn't causing failures earlier.